### DISCUSSION TITLE
I couldn't verify my Raffle Contract

### DISCUSSION THREAD

**Discussion by EngrPips** [Cyfrin/foundry-full-course-cu#230] [Q&A]
i have been trying to deploy and verify my raffle contract but i was only successful with deploying but not verifying i keep getting the below error

```
ONCHAIN EXECUTION COMPLETE & SUCCESSFUL.
Total Paid: 0.00461716501539055 ETH (1539055 gas * avg 3.00000001 gwei)
##
Start verification for (2) contracts
Start verifying contract `0x01593a8fde831ffa77e2736732851f92e4a352e8` deployed on sepolia

Submitting verification for [lib/foundry-devops/src/DevOpsTools.sol:DevOpsTools] "0x01593a8fdE831FfA77E2736732851f92e4a352E8".

Submitting verification for [lib/foundry-devops/src/DevOpsTools.sol:DevOpsTools] "0x01593a8fdE831FfA77E2736732851f92e4a352E8".

Submitting verification for [lib/foundry-devops/src/DevOpsTools.sol:DevOpsTools] "0x01593a8fdE831FfA77E2736732851f92e4a352E8".
Submitted contract for verification:
        Response: `OK`
        GUID: `btz67zj22plwgk2ns4dmpkuh67mckcvrkvtqsgqnewvffd8kjg`
        URL:
        https://sepolia.etherscan.io/address/0x01593a8fde831ffa77e2736732851f92e4a352e8
Contract verification status:
Response: `NOTOK`
Details: `Pending in queue`
Contract verification status:
Response: `NOTOK`
Details: `Other Exception - Please contact us for more information.`
Contract failed to verify.
make: *** [deploy] Error 1
```

I am not sure what to do, i would greatly appreciate any help at all @PatrickAlphaC  @alymurtazamemon

**Comment by pyromaus** [comment]
Make sure you have an Etherscan API Key in your `.env` and run `source .env` again 

This command might work:

`forge verify-contract --chain-id 11155111 --constructor-args $(cast abi-encode "constructor(uint256,uint256,address,bytes32,uint64,uint32)" <your entrance fee> <your interval> <your vrf coordinator> <your gas lane> <your sub Id> <your callback gas limit>) --etherscan-api-key $ETHERSCAN_API_KEY <your contract address> src/Raffle.sol:Raffle`

If not, run `foundryup` and try again.

**Reply by EngrPips** [reply]
Hello, thanks for your response , i truly appreciate it. So i ran the command you suggested and below was the response i got at the terminal

```
loyalty@Loyaltys-MBP foundry-smart-contract-lottery-f23 % forge verify-contract --chain-id 11155111 --constructor-args $(cast abi-encode "constructor(uint256,uint256,address,bytes32,uint64,uint32)" 0.01 ether  30 0x8103B0A8A00be2DDC778e6e7eaa21791Cd364625 0x474e34a077df58807dbe9c96d3c009b23b3c6d0cce433e59bbf5b34f823bc56c 1879 500000 --etherscan-api-key $ETHERSCAN_API_KEY 0x5B8D2FD73530e26D26e20E45aa685DfF4B53BA9a src/Raffle.sol:Raffle

cmdsubst> 
```
@pyromaus

**Reply by pyromaus** [reply]
for entrance fee, try `10000000000000000` don't include `ether`
there's a double space after `ether` --> only use one space.
the reason you're getting `cmdsubst>` is you forgot a closing bracket after `500000`

**Comment by shikhar229169** [comment]
Delete the cache folder and out folder and try again.

**Reply by EngrPips** [reply]
Thank you so much @shikhar229169 for your response, i truly appreciate it, i tried your suggestion and i still got the same error. not sure what to do now.

**Reply by shikhar229169** [reply]
Actually DevOpsTools is creating the problem, consider deleting the dev ops tool library, and remove all the functions related to DevOpsTools in interactions file or where ever you used it, and try again.

**Comment by jojomensah89** [comment]
Did anyone find a fix to this?

**Reply by Macbeth98** [reply]
As @shikhar229169 said, the issue was because of the ```DevOpsTools``` library. I removed it from the interactions file and changed the code a little to get the latest deployment of the ```Raffle``` contract by sending it as a constructor argument for the ```AddConsumer``` contract in Interactions.

Then this worked fine and my contract was successfully verified.
